src/Exporter/DataSource.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Exporter;
+
+use Doctrine\ORM\Query;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Exporter\DataSourceInterface;
+use Sonata\DoctrineORMAdminBundle\Datagrid\OrderByToSelectWalker;
+use Sonata\Exporter\Source\DoctrineORMQuerySourceIterator;
+use Sonata\Exporter\Source\SourceIteratorInterface;
+
+class DataSource implements DataSourceInterface
+{
+    public function createIterator(ProxyQueryInterface $query, array $fields): SourceIteratorInterface
```

Comment:
Psalm commented on this line, yet it doesn't fail? :thinking: What's wrong with Psalm?

Reply:
previously it was called `$field`, but I fixed it in SonataAdmin